public/js/widgets/address-autocomplete.js

```diff
@@ -0,0 +1,146 @@
+let stylesInitialized = false;
+
+class AddressAutocomplete extends HTMLElement {
+	constructor()
+	{
+		super();
+		this.input = null;
+		this.wrapper = null;
+		this.row = null;
+		this.autocompleteChoices = null;
+		this.api = '';
+		this.getPredictionsRoute = Routing.generate('bgm_ajax_get_address_predictions', {});
+		this.getDetailsRoute = Routing.generate('bgm_ajax_get_address_details', {});
+
+		this.init();
+	}
+
+	init()
+	{
+		this.input = this.querySelector('input');
+		this.wrapper = this;
+		this.row = this.parentNode.parentNode;
+		this.autocompleteChoices = this.querySelector('.autocomplete-choices');
+		this.api = this.dataset.api;
+
+		this.querySelector('.loading-overlay').setAttribute('style', '');
+
+		this.initEvents();
+		this._injectCSS();
+	}
+
+	initEvents()
+	{
+		let userInputDelay = null;
+		// get address predictions based on the user's input
+		this.input.addEventListener('input', () => {
+			clearTimeout(userInputDelay);
+			// prevent the request from being sent if value is empty
+			if (this.input.value == "") {
+				return;
+			}
+			// Add a timeout to prevent multiple requests being sent while the user is typing
+			userInputDelay = setTimeout(() => {
+				this.classList.add('loading');
+
+				fetch(`${this.getPredictionsRoute}?search=${this.input.value}&api=${this.api}`, {method: 'GET'})
+					.then(response => response.json())
+					.then(choices => { 
+						this.classList.remove('loading');
+						this.autocompleteChoices.innerHTML = '';
+
+						this.generateChoices(choices);
+					});
+			}, 800);
```

Comment:
J'aime le debounce, mais 800ms ça me semble un peu long. 
Ça doit sembler laggy / janky non?

Reply:
Aussi: tu peux toujours canceller la requête ajax précédente avec un [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) si tu veux.

Reply:
Good to know pour la prochaine fois que je vais créer quelque chose du genre !

Je l'ai réduit à 500 pour éviter que ça soit trop long :)

Par contre j'ai testé avec ma vitesse de tape et il lance plusieurs requêtes pendant que je tape. Je crois pas que ça soit dramatique étant donné qu'il charge à la session et non au call